vision.md

```diff
@@ -0,0 +1,41 @@
+# Buildfarmer Payload tasks process vision
+Subset of tasks defined to increase the quality of the ROS/Gazebo code by actively tracking and categorizing problems shipped with new code in a timely manner, making it easier for the development teams to:
+* Fix most errors when they are introduced.
+* Decide if an error that pops up in CI is a flaky test already existing in the codebase.
+* Identify how often an error occurs and decide how to proceed.
+
+## Internal process and rationale behind it
+
+We already have a priority document explaining the order of relevancy of the possible errors, this is a new set of standards to define what we expect of the internal buildfarmer process:
+
+1. Auto-log any error appearing in the nightlies, even when there is no one around. This could serve future investigations.
+2. Any person can be assigned to Buildfarmer Payload in case of necessity, this implies:
+   * Basic duties should be easy to automate, even scripted.
+   * Everyone assigned to Buildfarmer Payload must log/communicate internally any actions completed in the name of the role, so other folks can make progress on top of previous actions.
+   * There will be a public log with investigations related to the Buildfarmer Payload duties.
+   * Actions related to Buildfarmer Payload will be differentiated from actions related to the infra team, it is not resposability of the role to investigate infra issues, instead those will be escalated.
+3. Core Team and community developers should be able to check if an existent error in his build it’s also happening in the nightlies CI.
+4. Consecutive test failures priorities policy: Test errors occuring consecutively will be prioritized in terms of order of investigation, in the case of Gazebo jobs, the failing build will be retriggered to confirm if the error occurs a third time in a row. Errors happening three times in a row will be reported immediatly to the development teams in case the buildfarmer investigation wasn't enough to solve it. Errors that have appeared at least three times every 2 weeks, will also be prioritized. Buildfarmer Payload scripts will prioritize accordingly new investigations based on this.
```

Comment:
```suggestion
4. Consecutive test failures priorities policy: Test errors occuring consecutively will be prioritized in terms of order of investigation. In the case of Gazebo jobs, the failing build will be retriggered to confirm if the error occurs a third time in a row. Errors happening three times in a row will be reported immediately to the development teams in case the buildfarmer investigation wasn't enough to solve it. Errors that have appeared at least three times in a 2 week window, will also be prioritized. Buildfarmer Payload scripts will prioritize accordingly new investigations based on this.
```

Not sure if my last edit conveys what you meant.

Reply:
Yes! thank you